frontend/tests/controllers/ContestUpdateTest.php

```diff
@@ -419,24 +435,105 @@ public function testUpdateWindowLength() {
 
         $this->assertNull($contest['window_length'], 'Window length is not setted, because 0 is not a valid value');
 
-        $r['window_length'] = 60;
+        $windowLength = 10;
+        $r['window_length'] = $windowLength;
+        // Call API
+        $response = ContestController::apiUpdate($r);
+
+        $contest = ContestController::apiDetails($r);
+
+        $this->assertEquals($windowLength, $contest['window_length']);
+
+        // Update time for testing
+        Time::setTimeForTesting(Time::get() + 700);
+
+        try {
+            // Trying to create a run out of contest time
+            RunsFactory::createRun($problemData, $contestData, $contestant);
+            $this->fail('User could not create a run when is out of contest time');
+        } catch (NotAllowedToSubmitException $e) {
+            // Pass
+            $this->assertEquals('runNotInsideContest', $e->getMessage());
+        }
+
+        $windowLength = 40;
+        $r['window_length'] = $windowLength;
         // Call API
         $response = ContestController::apiUpdate($r);
 
         $contest = ContestController::apiDetails($r);
 
-        $this->assertEquals(60, $contest['window_length']);
+        $this->assertEquals($windowLength, $contest['window_length']);
+
+        // Trying to create a run inside contest time
+        $runData = RunsFactory::createRun($problemData, $contestData, $contestant);
 
         $r['window_length'] = 'Not valid';
 
         try {
             // Call API
-            $response = ContestController::apiUpdate($r);
+            ContestController::apiUpdate($r);
             $this->fail('Only numbers are allowed in window_length field');
         } catch (InvalidParameterException $e) {
             // Pass
             $this->assertEquals('parameterNotANumber', $e->getMessage());
         }
+
+        $identities = ContestController::apiUsers(new Request([
+            'auth_token' => $directorLogin->auth_token,
+            'contest_alias' => $contestData['request']['alias'],
+        ]));
+
+        $index = array_search($contestant->username, array_column($identities['users'], 'username'));
```

Comment:
nit: estas líneas largas puedes partirlas?

```suggestion
        $index = array_search(
            $contestant->username,
            array_column($identities['users'], 'username')
        );
```

igual en L505, L508 y las otras líneas similares.

Reply:
Listo